gnovm/pkg/gnolang/values.go

```diff
@@ -1519,6 +1519,7 @@ func (tv *TypedValue) AssertNonNegative(msg string) {
 }
 
 func (tv *TypedValue) ComputeMapKey(store Store, omitType bool) MapKey {
+	fillValueTV(store, tv)
```

Comment:
this isn't the right place; tv should already be filled, and children should be filled if needed.

Reply:
see later recursive usage of ComputeMapKey also, where fillValueTV is called before.

the problem is in realm.go:
```go
	case *MapValue:
		cv.vmap = make(map[MapKey]*MapListItem, cv.List.Size)
		for cur := cv.List.Head; cur != nil; cur = cur.Next {
			fillTypesTV(store, &cur.Key). <--- should fill value too. 
			fillTypesTV(store, &cur.Value)

			cv.vmap[cur.Key.ComputeMapKey(store, false)] = cur
		}
		return cv

```

I don't think this is a problem with the interrealm refactor.